src/Microsoft.Fast.Components.FluentUI/Components/Icon/FluentIcon.cs

```diff
@@ -201,6 +240,56 @@ private string ComposedName
         }
     }
 
+    private bool GetIconChanged()
+    {
+        return _previousIconUrl != _iconUrl;
+    }
+
+    private bool GetParametersChanged()
+    {
+        if (_previousId != Id ||
+            _previousColor != _color ||
+            _previousSlot != Slot ||
+            _previousClass != Class ||
+            _previousStyle != Style ||
```

Comment:
Just for the sake to be explicit about checking parameters and avoid extra allocations when style is build. If later is not a big deal from your point of view, i'll replace with StyleValue - it will also eliminate _color/_previousColor check

Reply:
Make sense. Thanks.